server/src/main/java/io/deephaven/server/table/ops/ApplyPreviewColumnsGrpcImpl.java

```diff
@@ -17,14 +18,16 @@
 @Singleton
 public class ApplyPreviewColumnsGrpcImpl extends GrpcTableOperation<ApplyPreviewColumnsRequest> {
     @Inject
-    protected ApplyPreviewColumnsGrpcImpl() {
-        super(BatchTableRequest.Operation::getApplyPreviewColumns, ApplyPreviewColumnsRequest::getResultId,
-                ApplyPreviewColumnsRequest::getSourceId);
+    protected ApplyPreviewColumnsGrpcImpl(final TableServiceContextualAuthWiring authWiring) {
+        super(authWiring::checkPermissionApplyPreviewColumns, BatchTableRequest.Operation::getApplyPreviewColumns,
+                ApplyPreviewColumnsRequest::getResultId, ApplyPreviewColumnsRequest::getSourceId);
     }
 
     @Override
-    public Table create(ApplyPreviewColumnsRequest request, List<SessionState.ExportObject<Table>> sourceTables) {
+    public Table create(final ApplyPreviewColumnsRequest request,
+            final List<SessionState.ExportObject<Table>> sourceTables) {
         Assert.eq(sourceTables.size(), "sourceTables.size()", 1);
+        checkPermission(request, sourceTables);
```

Comment:
Thoughts on having checkPermission be the very first part of create() (generally)?